src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs

```diff
@@ -451,29 +452,59 @@ public static unsafe Vector64<uint> Create(uint e0, uint e1)
         /// <returns>A new <see cref="Vector64{Byte}" /> instance with the first element initialized to <paramref name="value"/> and the remaining elements initialized to zero.</returns>
         public static unsafe Vector64<byte> CreateScalar(byte value)
         {
-            var result = Vector64<byte>.Zero;
-            Unsafe.WriteUnaligned(ref Unsafe.As<Vector64<byte>, byte>(ref result), value);
-            return result;
+            if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.Insert(Vector64<byte>.Zero, 0, value);
+            }
+
+            return SoftwareFallback(value);
```

Comment:
Curious about wrapping the `SoftwareFallback()` in a static method. What is the advantage of doing it?

Reply:
Not sure, actually. I did this to be consistent with existing Vector128/256 implementations. 
@tannergooding Do you know why?

Reply:
With the logic for all the various paths the method is too large for the normal inlining heuristics to work so we need to mark it `AggressiveInlining` (since the accelerated paths will generally be pretty small).
However, we don't necessarily want the SoftwareFallback to be inlined as that may not be beneficial.
Putting it in its own method prevents it from being inlined in the normal case and allows the JIT to decide if it is "too large or not" by itself.

Reply:
That make sense.